src/render.c

```diff
@@ -70,7 +70,12 @@ void render_con(Con *con, bool render_fullscreen, bool already_inset) {
         }
         inset.height = 0;
         if (con->deco_rect.width != 0 && con->deco_rect.height != 0) {
-            con->deco_rect = rect_add(con->deco_rect, inset);
+            if (con_is_leaf(con)) {
+                con->deco_rect.x = con->deco_rect.y = 0;
```

Comment:
This doesn't strike me as correct. `deco_rect` has coordinates relative to the parent's `rect`. In split windows, for example, the `x` / `y` values do not have to be 0.

Reply:
Oh yeah, you're right. That line shouldn't be there.